main.nf

```diff
@@ -635,111 +641,245 @@ process VCF_CONSENSUS {
 
-        BCFTOOLS_FILTER(FREEBAYES.out.variant)
-    
-    } else {
-        
-        BCFTOOLS_FILTER(FREEBAYES.out.variant)
-    }
-     
-    //----------------------------------------
-    // Annotate Genomic Variants    
-    //----------------------------------------
-
-    if (params.prediction == 'snpeff') {
-        //deprecated until further notice        
-        SNPEFF(BCFTOOLS_FILTER.out.variant, 
-               FASTP.out.reads)
-    
-    } else {
+//----------------------------------------
+// WORKFLOW: bowtie2
+// Takes:
+//      ref = reference genome
+//      reads = paired end reads
+//      phix = phix genome
+//      
+// Main: 
+//      1. Quality check and read filtering 
+//      2. Read mapping using BOWTIE2
+//      3. Variant calling
+//      4. Consensus 
+//
+//----------------------------------------
+workflow bowtie2 {
+
+    take: 
+        ref
+        reads
+        phix
+
+    main:
+        quality_check(reads, phix)
+        BOWTIE2(ref, quality_check.out.reads)
+        variants(ref, BOWTIE2.out.align)
+        consensus(variants.out.variant, reads, ref)
+}
+//----------------------------------------
+
+//----------------------------------------
+// WORKFLOW: minimap2
+// Takes:
+//      ref = reference genome
+//      reads = paired end reads
+//      phix = phix genome
+//      
+// Main: 
+//      1. Quality check and read filtering 
+//      2. Read mapping using MINIMAP2
+//      3. Variant calling
+//      4. Consensus 
+//
+//----------------------------------------
+workflow minimap2 {
+
+    take: 
+        ref
+        reads
+        phix 
+
+    main:
+        quality_check(reads, phix)
+        MINIMAP2(ref, quality_check.out.reads)
+        variants(ref, MINIMAP2.out.align)
+        consensus(variants.out.variant,reads, ref)
+}
+//----------------------------------------
 
-        SNIPPY(BCFTOOLS_FILTER.out.variant, 
-               FASTP.out.reads)
+//----------------------------------------
+// WORKFLOW: quality_check
+// Takes:
+//      reads = paired end reads
+//      phix = phix genome
+//      
+// Main: 
+//      1. Filter using phix  
+//      2. Trim paired end reads  
+//      3. Check quality trimmed pe reads
+//
+// Emit: 
+//      reads = Trimmed paired end reads
+//
+//----------------------------------------
+workflow quality_check {
 
-    }
+    take:
+        reads
+        phix
     
-    //----------------------------------------
-    // Consensus building 
-    //----------------------------------------
+    main:
+        
+        REMOVE_PHIX(phix, reads)
+        FASTP(REMOVE_PHIX.out.reads)
+        fastqc_reports = FASTQC(FASTP.out.reads)
     
-    if (params.consensus == 'bcftools') {
+    emit: 
+        reads = FASTP.out.reads
+}
+//----------------------------------------
 
-        BCFTOOLS_CONSENSUS(ch_ref, SNIPPY.out.annotation)
-    
-    } else {
+workflow {
+    main:    
+        ref = Channel.value(file("${params.ref}"))
```

Comment:
This should be 
```nextflow
ref = Channel.fromPath( params.ref )
```
to handle multiple reference genome FASTA files being passed.

See issue #15

Reply:
This has been addressed and reference in a commit and merge.